docs/sophia_features.md

```diff
@@ -482,6 +482,25 @@ function g(p : int * option(int)) : int =
   b
 ```
 
+Guards are boolean functions that can be used on patterns in both switch
```

Comment:
I'd elaborate a little bit on that. Note that many smart contract devs aren't familiar with functional programming. Also, mention that these expressions cannot be stateful

Reply:
Let me know if you think this makes it clearer 2deb568be3081db48f2ea0ddf77337aa1bde88ba

Reply:
Done that here 70af4828483be4a8ac93a9b123411e2e0b946fef